js/modifyRecurringContributionPage.js

```diff
@@ -0,0 +1,50 @@
+CRM.$(function () {
+  const payment_scheme_schedule = CRM.vars.membershipextras.payment_scheme_schedule;
+  hideUnnecessaryPaymentPlanFields();
+  addFuturePaymentSchemeSchedule();
+
+  function hideUnnecessaryPaymentPlanFields() {
+    const frequency = CRM.vars.membershipextras.recur_contribution['frequency_unit'];
+    const payment_scheme_fields_to_hide = ['Cycle Day', 'Frequency', 'Installments', 'Next Contribution'];
+
+    CRM.$('.crm-recurcontrib-view-block > table > tbody > tr').each(function() {
+      // hide 'Cycle Day' for all annual payment plans.
+      if(CRM.$('td.label', this).text() === 'Cycle Day' && frequency === 'year') {
+        CRM.$(this).hide();
+      }
+
+      if(payment_scheme_fields_to_hide.includes(CRM.$('td.label', this).text()) && payment_scheme_schedule !== null) {
+        CRM.$(this).hide();
+      }
+
+      if(CRM.$('td.label', this).text() === 'Next Contribution') {
+        CRM.$('td.label', this).text('Next Scheduled Contribution Date');
+      }
+    });
+  }
+
+  function addFuturePaymentSchemeSchedule() {
+    if (payment_scheme_schedule === null) {
+      return;
+    }
+
+    var payment_scheme_block = '<tr><td class="label">Future payment scheme Instalments</td><td>';
```

Comment:
Don't we localise these texts? 

Perhaps, it is better if we put the table in the smarty template, so it is faster than concatenating string in browser the content on run time?

Reply:
> Don't we localise these texts?

While this is true, for all the time I've been here at Compuco we never had to do translation for any client nor any one ever created a github issue asking for translation support or anything, so I am happy to skip it.

> Perhaps, it is better if we put the table in the smarty template, so it is faster than concatenating string in browser the content on run time?

I really don't see that there will be  much of performance improvement (if any) to justify changing the approach.